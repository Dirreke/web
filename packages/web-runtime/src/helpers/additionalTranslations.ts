// just a dummy function to trick gettext tools
function $gettext(msg: string): string {
  return msg
}

export const additionalTranslations = {
  fileNotAvailable: $gettext('The requested file is not yet available, please try again later.'),
  activities: $gettext('Activities'),
  noActivities: $gettext('No activities'),
  virusDetectedActivity: $gettext(
    'Virus "%{description}" detected. Please contact your administrator for more information.'
  )
}
