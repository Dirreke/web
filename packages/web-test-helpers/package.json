{
  "name": "web-test-helpers",
  "private": true,
  "peerDependencies": {
      "@vue/composition-api": "1.7.1",
      "vue-gettext": "^2.1.5",
      "vue-router": "3.6.5",
      "vuex": "^3.6.2"
  }
}
