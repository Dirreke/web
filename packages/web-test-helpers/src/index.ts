export * from './mocks/defaultComponentMocks'
export * from './mocks/defaultStubs'
export * from './mocks/axios'

export * from './defaultPlugins'
export * from './helpers'
