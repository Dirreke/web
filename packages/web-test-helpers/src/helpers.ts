import { defaultLocalVue } from 'web-test-helpers/src/localVue/defaultLocalVue'
import Vuex, { StoreOptions } from 'vuex'
import { Component } from 'vue'
import { mount as _mount } from '@vue/test-utils'
import { Data, defineComponent, SetupFunction } from '@vue/composition-api'
import { defaultPlugins } from 'web-test-helpers'
export { RouterLinkStub } from '@vue/test-utils'

export const createStore = <T>(storeOptions: StoreOptions<T>) => {
  return {
    install(app) {
      app.prototype.$store = new Vuex.Store(storeOptions)
    }
  }
}

type ComponentType = any

type CompatPlugin = any

type CompatGlobalMountOptions = {
  mocks?: object | false
  stubs?:
    | {
        [key: string]: Component | string | boolean
      }
    | string[]
  attrs?: Record<string, unknown>
  plugins?: (CompatPlugin | [CompatPlugin, ...any[]])[]
  directives?: { [key: string]: any }
  provide?: Record<any, any>
  components?: Record<string, ComponentType>
}

type CompatMountOptions = {
  props?: any
  data?: any
  attachTo?: HTMLElement | string
  global?: CompatGlobalMountOptions
  shallow?: boolean
  slots?: any
}

export const mount = (component: ComponentType, options: CompatMountOptions) => {
  const localVue = defaultLocalVue()

  options?.global?.plugins?.filter((p) => p).forEach(localVue.use)

  for (const [name, component] of Object.entries(options?.global?.components || {})) {
    localVue.component(name, component)
  }

  return _mount(component, {
    localVue,
    ...(options?.shallow && { shouldProxy: options.shallow }),
    ...(options?.props && { propsData: options.props }),
    ...(options?.data && { data: options.data }),
    ...(options?.attachTo && { attachTo: options.attachTo }),
    ...(options?.slots && { slots: options.slots }),
    ...(options?.global?.provide && { provide: options.global.provide }),
    ...(options?.global?.mocks && { mocks: options.global.mocks }),
    ...(options?.global?.stubs && { stubs: options.global.stubs })
  })
}

export const shallowMount = (component: ComponentType, options: CompatMountOptions) => {
  options = options || {}
  options.shallow = true

  return mount(component, options)
}

export const getComposableWrapper = (
  setup: SetupFunction<Data, Data>,
  { mocks = undefined, store = undefined } = {}
) => {
  return mount(
    defineComponent({
      setup,
      template: '<div></div>'
    }),
    {
      global: {
        plugins: [...defaultPlugins(), store],
        ...(mocks && { mocks })
      }
    }
  )
}
