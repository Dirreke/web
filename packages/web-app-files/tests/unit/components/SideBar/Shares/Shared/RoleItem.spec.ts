import RoleItem from 'web-app-files/src/components/SideBar/Shares/Shared/RoleItem.vue'
import { mockDeep } from 'jest-mock-extended'
import { ShareRole } from 'web-client/src/helpers/share'
import { defaultPlugins, mount } from 'web-test-helpers'

describe('RoleItem', () => {
  it('renders the role information', () => {
    const { wrapper } = createWrapper()
    expect(wrapper).toMatchSnapshot()
  })
})

function createWrapper({ props = {} } = {}) {
  return {
    wrapper: mount(RoleItem, {
      props: { role: mockDeep<ShareRole>(), allowSharePermission: true, ...props },
      global: { plugins: [...defaultPlugins()] }
    })
  }
}
