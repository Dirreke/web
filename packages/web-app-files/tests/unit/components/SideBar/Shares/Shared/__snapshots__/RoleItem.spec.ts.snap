// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoleItem renders the role information 1`] = `
<span class="roles-select-role-item" id="files-role-undefined">
  <span class="oc-text-bold oc-display-block oc-width-1-1">undefined</span>
  <span class="oc-m-rm oc-text-small oc-display-block"></span>
</span>
`;
